src/neomaril_codex/external_monitoring.py

```diff
@@ -0,0 +1,526 @@
+            },
+            timeout=60,
+        )
+        formatted_msg = parse_json_to_yaml(response.json())
+
+        if response.status_code == 201:
+            logger.debug(
+                f"External monitoring was successfully registered:\n{formatted_msg}"
+            )
+            external_monitoring_hash = response.json()["ExternalMonitoringHash"]
+            return external_monitoring_hash
+
+        if response.status_code == 401:
+            logger.debug(
+                "Login or password are invalid, please check your credentials."
+            )
+            raise AuthenticationError("Login not authorized.")
+
+        if response.status_code == 404:
+            logger.debug("Group not found in the database")
+            raise GroupError("Group not found in the database")
+
+        if response.status_code >= 500:
+            logger.debug("Server is not available. Please, try it later.")
+            raise ServerError("Server is not available!")
+
+        logger.debug(f"Something went wrong...\n{formatted_msg}")
+        raise ExternalMonitoringError("Could not register the monitoring.")
+
+    def register_monitoring(
+        self,
+        *,
+        configuration_file: Union[str, dict],
+        group: Optional[str] = "datarisk",
+        model_file: Optional[str] = None,
+        requirements_file: Optional[str] = None,
+        preprocess_file: Optional[str] = None,
+        preprocess_reference: Optional[str] = None,
+        shap_reference: Optional[str] = None,
+        python_version: Optional[str] = "3.10",
+        wait_ready: Optional[bool] = True,
+    ) -> NeomarilExternalMonitoring:
+        """Register and host a Neomaril External Monitoring
+
+        Args:
+            configuration_file (Union[str, dict]): Path or dict that represents the configuration file
+            group (Optional[str]): Group the model is inserted. Default is 'datarisk' (public group). Defaults to "datarisk".
+            model_file (Optional[str]): Path to your locally trained model. Defaults to None.
+            requirements_file (Optional[str]): Path of the requirements file. Defaults to None.
+            preprocess_file (Optional[str]): Path of the preprocess script. Defaults to None.
+            preprocess_reference (Optional[str]): Preprocessing function entrypoint. Defaults to None.
+            shap_reference (Optional[str]): Shap function entrypoint. Defaults to None.
+            python_version (Optional[str]): Python version. Can be "3.8", "3.9" or "3.10". Defaults to "3.10".
+            wait_ready (Optional[bool]): Wait external monitoring be ready. Defaults to True.
+
+        Returns:
+            NeomarilExternalMonitoring
+        """
+
+        base_external_url = f"{self.base_url}/external-monitoring/{group}"
+
+        external_monitoring_hash = self.__register(
+            configuration_file=configuration_file, url=base_external_url
```

Comment:
I think we can get this as fields instead of a json file. Its easier for the user edit. We could even change this later in the old monitoring.

Reply:
Maybe ask a DS what they want (Gabriel Bueno and Joao Victor just used the monitoring)

Reply:
Actually I'm using a dict instead of a json file to test:
```python
configuration_file = {
    "Name": "Teste",
    "TrainingExecutionId": 1,
    "Period": "Week",
    "InputCols": [
        "VALOR_A_PAGAR", "TAXA", "RENDA_MES_ANTERIOR", "NO_FUNCIONARIOS",
        "RZ_RENDA_FUNC", "VL_TAXA","DDD", "SEGMENTO_INDUSTRIAL",
        "DOMINIO_EMAIL", "PORTE", "CEP_2_DIG"],
    "OutputCols": [ "probas" ],
    "DataSourceName": "TestAzure",
    "ExtractionType": "Full",
    "DataSourceUri": "https://samplesneomaril.blob.core.windows.net/data/risk_model.parquet",
    "ColumnName": "SAFRA_REF"
}
```

I was thinking about add in the ExternalMonitoring.ipynb a cell opening and loading a json file as well and other cell with the dict constructed by hand. What do you think about it?

Reply:
The point is, i think the multiple parameters are easier to understand and document instead of one parameter that is a dict with mandatory keys.

Reply:
Also, you are not opening the file when the user sends a str path